pkg/util/errors.go

```diff
@@ -63,22 +63,14 @@ func WriteError(err error, writer http.ResponseWriter) {
 
 // HandleResponseError builds at HttpErrorResponse from the given response.
 func HandleResponseError(response *http.Response) error {
-	logger := log.D()
-	logger.Errorf("Handling failure response: returned status code %d", response.StatusCode)
-	httpErr := &HTTPError{
-		StatusCode: response.StatusCode,
-	}
-
 	body, err := BodyToBytes(response.Body)
 	if err != nil {
 		return fmt.Errorf("error processing response body of resp with status code %d: %s", response.StatusCode, err)
 	}
 
-	if err := BytesToObject(body, httpErr); err != nil || httpErr.Description == "" {
-		logger.Debugf("Failure response with status code %d is not an HTTPError. Error converting body: %v. Default err will be returned.", response.StatusCode, err)
-		return fmt.Errorf("StatusCode: %d Body: %s", response.StatusCode, body)
-	}
-	return httpErr
+	err = fmt.Errorf("StatusCode: %d Body: %s", response.StatusCode, body)
+	log.D().Errorf("Call to client failed with: %s", err)
```

Comment:
the `response.Request` should be populated if this is a client request so this can be logged with context